src/ui/public/saved_objects/saved_objects_client.js

```diff
@@ -232,6 +232,16 @@ export class SavedObjectsClient {
       return Promise.reject(new Error('body not permitted for GET requests'));
     }
 
-    return kfetch({ method, pathname: path, query, body: JSON.stringify(body) });
+    return kfetch({ method, pathname: path, query, body: JSON.stringify(body) })
+      .catch(resp => {
```

Comment:
The problem is that you are not getting the REST `resp` from kfetch. kfetch is going to reject the promise with a [FetchError](https://github.com/elastic/kibana/blob/master/src/ui/public/kfetch/kfetch.ts#L86)

Reply:
`FetchError` still has `res` and `body` properties:
https://github.com/elastic/kibana/blob/master/src/ui/public/kfetch/kfetch.ts#L29

I'll rename these variables so they reflect an instance of `FetchError`.